section3/reflection.md

```diff
@@ -1,17 +1,29 @@
 ## Section 3 Reflection
 
 1. What are two points from the Growth Mindset article and/or video that either resonated with you, or were brand new to you?
+* I have seen the SMART goals before, and I have found them helpful when lesson planning for others. I could see them being helpful here.
+* I liked the idea that if I am not encountering challenge, then I am probably not solving any problems.
 
 1. In which ways do you currently demonstrate a Growth Mindset? In which ways do you _not_?
+I am currently very intimidated by challenge in this format, and tend to shut down--the opposite of a growth mindset. On the other hand, I have stuck it out this far.
 
 1. What is a Hash, and how is it different from an Array?
+Hashes, short for hashmap, are data structures. They are lists, but instead of requiring the listed items to be referenced by number, they can be associated with pretty much any other value.
 
 1. In the space below, create a Hash stored to a variable named `pet_store`.  This hash should hold an inventory of items and the number of that item that you might find at a pet store.
+`pet_store = {puppies: 12, kittens: 8, birds: 6, rats: 4, snakes: 1}``
 
 1. Given the following `states = {"CO" => "Colorado", "IA" => "Iowa", "OK" => "Oklahoma"}`, how would you access the value `"Iowa"`?
+`states["IA"]``
 
 1. With the same hash above, how would we get all the keys?  How about all the values?
+```
+states.keys
+states.values
+```
 
 1. What is another example of when we might use a hash?  In your example, why is a hash better than an array?
+I could store my inventory of climbing gear with a hash. This would allow me to associate keys with more helpful values than simply their position in a list. For example, I could associate with quantity, brand, age, etc.
 
 1. What questions do you still have about hashes?
+I have a hard time picturing the application of the .keys or .values methods by themselves. I need to see some more examples.
```

Comment:
Oh for sure! There are plenty of use cases. Some I can think of
- If I am searching through large data sets or deeply nested hashes, I might check my keys or values as I develop and test
- What if I have a hash that is storing every State as a key and its abbreviation as a value? Then calling hash.keys would be nice for pulling every state by it's full name, same with .values and pulling all abbreviations
- Bigger uses wouldn't be to call the keys or values as a list of arrays, but instead to iterate through them using `.each_keys` or `.each_values` (You will learn more about iteration/enumerables in Mod 1/2)